tests/unit_tests/test_diamond.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```

Comment:
Perhaps you should remove this test file from the PR and we can refocus on this later?